Remove native_functions.yaml dependency from GridSample.{cpp,cu}

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67875
* #67874
* #67721
* #66980
* __->__ #66979

Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

peterbell10 committed:
Remove native_functions.yaml dependency from GridSample.{cpp,cu}

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 commented:
@dagitses ping

malfet commented:
Commandeering stack from @dagitses

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Fix overflow for 32-bit size_t on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]

peterbell10 commented on aten/src/ATen/TensorGeometry.cpp:
```diff
@@ -1,10 +1,30 @@
 
+// See TensorGeometry.h on why this is useful now that we cache is_contiguous.
+bool geometry_is_contiguous(IntArrayRef sizes, IntArrayRef strides) {
+  assert(sizes.size() < std::size_t{std::numeric_limits<std::int64_t>::max()});
```
Looking at this again, this isn't safe for 32-bit `size_t` where it would overflow. I've replaced it with `c10::overflows` instead.

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856103).

peterbell10 committed:
Rebase on master on "Remove native_functions.yaml dependency from GridSample.{cpp,cu}"


Differential Revision: [D31856103](https://our.internmc.facebook.com/intern/diff/D31856103)

[ghstack-poisoned]
